app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/handler/SwapStateInitialLoadingHandler.kt

```diff
@@ -0,0 +1,39 @@
+package org.p2p.wallet.swap.jupiter.statemanager.handler
+
+import org.p2p.wallet.swap.jupiter.domain.model.SwapTokenModel
+import org.p2p.wallet.swap.jupiter.repository.tokens.JupiterSwapTokensRepository
+import org.p2p.wallet.swap.jupiter.statemanager.SwapState
+import org.p2p.wallet.swap.jupiter.statemanager.SwapStateAction
+import org.p2p.wallet.swap.jupiter.statemanager.SwapStateManager.Companion.DEFAULT_SLIPPAGE
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class SwapStateInitialLoadingHandler(
+    private val jupiterTokensRepository: JupiterSwapTokensRepository,
+) : SwapStateHandler {
+
+    override fun canHandle(state: SwapState): Boolean = state is SwapState.InitialLoading
+
+    override suspend fun handleAction(
+        stateFlow: MutableStateFlow<SwapState>,
+        state: SwapState,
+        action: SwapStateAction
+    ) {
+        val oldState = state as SwapState.InitialLoading
+
+        when (action) {
+            is SwapStateAction.InitialLoading -> {
+                stateFlow.value = SwapState.InitialLoading
+
+                val jupiterTokens = jupiterTokensRepository.getTokens()
+                val first = jupiterTokens.find { it.tokenName == "SOL" } ?: return
```

Comment:
Constants.SOL_SYMBOL

Reply:
or better - compare by mints

Reply:
first -> tokenA
second -> tokenB

Reply:
it's for testing, remove here https://github.com/p2p-org/key-app-android/pull/1290